### PR TITLE
keynames: fix addWarning

### LINKED ISSUE #3577: output of more than 10 warnings broken
```
 Sorry, 111 warnings were issued ;(
        Sorry, module kdb issued the warning C01200:
        Installation: Dlopen failed. Could not load module libelektra-ini.so. Reason: libelektra-ini.so: cannot open shared object file: No such file or directory
        Sorry, module kdb issued the warning C01200:
        Installation: Could not load plugin ini in process plugin
        Sorry, module kdb issued the warning C01200:
        Installation: Method 'elektraProcessPlugins' for get failed
        Sorry, module kdb issued the warning C01310:
        Internal: Could not reference back to plugin system:/elektra/plugins/ini
        Sorry, module kdb issued the warning C01200:
        Installation: Dlopen failed. Could not load module libelektra-ini.so. Reason: libelektra-ini.so: cannot open shared object file: No such file or directory
        Sorry, module kdb issued the warning C01200:
        Installation: Could not load plugin ini in process plugin
        Sorry, module kdb issued the warning C01200:
        Installation: Method 'elektraProcessPlugins' for get failed
        Sorry, module kdb issued the warning C01310:
        Internal: Could not reference back to plugin system:/elektra/plugins/ini
        Sorry, module kdb issued the warning C01200:
        Installation: Dlopen failed. Could not load module libelektra-null.so. Reason: libelektra-null.so: cannot open shared object file: No such file or directory
        Sorry, module kdb issued the warning C01200:
        Installation: Could not load plugin null in process plugin
        Sorry, module The command kdb ls terminated unsuccessfully with the info:
key meta:/warnings/#_10/module was not found
Please report the issue at https://issues.libelektra.org/
```

Wild guess: It wants to iterate to `#_10` but actually `#10` would be correct?

### PR DESCRIPTION
**Pull request by kodebach**
closes #3577

## Basics

These points need to be fulfilled for every PR:

- [x] Short descriptions of your changes are in the release notes
      (added as entry in `doc/news/_preparation_next_release.md` which
      contains `_(my name)_`)
      **Please always add something to the release notes.**
- [x] Details of what you changed are in commit messages
      (first line should have `module: short statement` syntax)
- [x] References to issues, e.g. `close #X`, are in the commit messages.
- [x] The buildservers are happy.
- [x] The PR is rebased with current master.

## Checklist

- [x] I added unit tests for my code
- [x] I fully described what my PR does in the documentation
      (not in the PR description)
- [x] I fixed all affected documentation
- [x] I added code comments, logging, and assertions as appropriate (see [Coding Guidelines](https://master.libelektra.org/doc/CODING.md))
- [x] I updated all meta data (e.g. README.md of plugins and [METADATA.ini](https://master.libelektra.org/doc/METADATA.ini))
- [x] I mentioned every code not directly written by me in [THIRD-PARTY-LICENSES](doc/THIRD-PARTY-LICENSES)

## Review

Reviewers will usually check the following:

- [ ] Documentation is introductory, concise, good to read and describes everything what the PR does
- [ ] Examples are well chosen and understandable
- [ ] Code is conforming to [our Coding Guidelines](https://master.libelektra.org/doc/CODING.md)
- [ ] APIs are conforming to [our Design Guidelines](https://master.libelektra.org/doc/DESIGN.md)
- [ ] Code is consistent to [our Design Decisions](https://master.libelektra.org/doc/decisions)

### CODE REVIEW

**Review by markus2330 — APPROVED** [review]
Thank you very much for the quick fix! :sparkling_heart: 

Btw. you could extend the release notes by and by, then there is not so much to write at once :wink:

File: doc/news/_preparation_next_release.md [hunk 1]
```diff
@@ -43,7 +43,7 @@ If you rely on specific behaviour of Elektra's Keynames and have already taken t
 
 <!-- TODO (kodebach): finish -->
 
-. _(Klemens Böswirth)_
+.. _(Klemens Böswirth)_
```
Comment (mpranj) [inline doc/news/_preparation_next_release.md:46]: I added at least a hint so it's easier to improve the release notes later.

```suggestion
- Fix combining dot-dot parts and escapes in key names.  _(Klemens Böswirth)_
- Fix adding more than 10 warnings to a key. _(Klemens Böswirth)_
```

**Review by mpranj — APPROVED** [review]
Thank you so much! 🚀

File: doc/news/_preparation_next_release.md [hunk 2]
```diff
@@ -43,7 +43,8 @@ If you rely on specific behaviour of Elektra's Keynames and have already taken t
 
 <!-- TODO (kodebach): finish -->
 
-. _(Klemens Böswirth)_
+- Fix combining dot-dot parts and escapes in key names.  _(Klemens Böswirth)_
```
Comment (mpranj) [inline doc/news/_preparation_next_release.md:46]:
```suggestion
- Fix combining dot-dot parts and escapes in key names. _(Klemens Böswirth)_
```

**Comment by kodebach** [comment]
I actually, intentionally didn't really update the release notes. This PR and #3576 both fix issues introduced by #3555. There was no release after #3555 was merged and I don't see why we should mention a bugfix for a bug that was never in any release.

**Comment by mpranj** [comment]
Thank you so much for the fix! ❤️ 

Please adapt the release notes as you see fit.

**Pull request merged**